core/blockly.ts

```diff
@@ -165,6 +165,13 @@ import {resizeSvgContents as realResizeSvgContents, WorkspaceSvg} from './worksp
 import * as Xml from './xml.js';
 import {ZoomControls} from './zoom_controls.js';
 
+declare global {
```

Comment:
I don't really think we should do this, though I could probably be convinced if we had good reasons.

I think declaring this as a new global would also affect anyone who pulls in our types, which may be undesirable for them. We had already casted everything to string, which is more explicit and reduces the possibility of surprises (such as mistakenly trying to set an attribute to `null` only to have it set to `"null"`). So I don't see a compelling reason to add this.

Reply:
This change strengthens our typing and does the opposite of what you describe.

1. This only opens the ability to set attributes to numbers (which comprise the majority of our calls to setAttribute).
2. This *reduces* the number of times we have to cast a value to a string, which means that if a 'number' is actually null or undefined, it will fail.
3. Casting everything to a string is much less explicit, since everything can be cast to a string.  This change removes massive numbers of unchecked, blind casts.

Reply:
What I meant by explicit is that anything passed to `setAttribute` will be implicitly cast to a string by the browser. Having to cast it to a string ourselves first is the explicit action we take. You don't have to know that the value you pass in will be cast to a string by the browser, because it's right there in our code. So I disagree with point 3, because the blind casts are still there, just hidden a layer deeper.

However, you're right that removing the casts in favor of checking for numbers (and possibly booleans) still provides us type-safety against unexpected `null` or `undefined` values which may otherwise just be cast to a string by our indiscriminate casting. 

I still don't really like that we would also be essentially opting in all of our downstream users into this change, and I don't know of a way around that.

Reply:
> I still don't really like that we would also be essentially opting in all of our downstream users into this change, and I don't know of a way around that.

I agree that this `declare global` is very undesirable for exactly that reason.

What about providing (and consistently using) a helper `function setAttribute(Element, name: string, value: number)`—perhaps in `utils/dom.ts`?